.github/workflows/solidity-foundry-artifacts.yml

```diff
@@ -0,0 +1,479 @@
+name: Solidity Foundry Artifact Generation
+on:
+  workflow_dispatch:
+    inputs:
+      product:
+        type: choice
+        description: 'product for which to generate artifacts; should be the same as Foundry profile'
+        required: true
+        options:
+          - "automation"
+          - "ccip"
+          - "functions"
+          - "keystone"
+          - "l2ep"
+          - "liquiditymanager"
+          - "llo-feeds"
+          - "operatorforwarder"
+          - "shared"
+          - "transmission"
+          - "vrf"
+      base_ref:
+        description: 'commit or tag to be used as base reference, when looking for modified Solidity files'
+        required: true
+
+env:
+  FOUNDRY_PROFILE: ci
+  # Has to match the `make foundry` version.
+  FOUNDRY_VERSION: nightly-de33b6af53005037b463318d2628b5cfcaf39916
```

Comment:
To avoid the inevitable drift, I wonder if should make this version value a variable within `contracts/GNUmakefile` and then add a step in this workflow to extract the version from that vile/var and then set this env var via GITHUB_ENV as a step in this workflow.

Reply:
nice idea!